episodes/2-keras.Rmd

```diff
@@ -312,6 +320,16 @@ seed(1)
 keras.utils.set_random_seed(2)
 ```
 
+::: callout
+## When to use random seeds?
+We use a random seed here to ensure that we get the same results every time we run this code.
+This makes our results reproducible and allows us to better compare results between different experiments.
+
+Please note that even though you have selected a random seed, this seed is used to generate a
+**different** random number every time you execute a Jupyter cell.
+So, to get truly replicable deep learning pipelines you need to run the notebook from start to end in one go.
+:::
```

Comment:
I have seen some confusion about this in the past, so perhaps we can make it even more explicit:
```suggestion
When the same code is run on another machine, the random number and hence the results can also differ.
:::
```

Reply:
@carschno I think this is actually not true. Or at least not the promise of using random seeds right? The whole idea of using random seeds is that the code becomes deterministic and the results are the same on another machine. Of course you should always double check that.

I agree that often students have a different result, but that is probably due to running cells multiple times or different dependency versions.

Reply:
As discussed with @carschno we decided to leave it like was proposed in this PR.